src/core/comm/coll.cc

```diff
@@ -50,18 +50,15 @@ enum CollTag : int {
 
 static int mpi_tag_ub = 0;
 
-static std::vector<MPI_Comm> mpi_comms;
+static std::vector<MPI_Comm> mpi_comms(0);
```

Comment:
this feels like an unnecessary micro-optimization that doesn't do any good. I wouldn't mind keeping around the default allocation for this vector even when no cpu communicators are used.

Reply:
I just want to make sure the vector is empty.

Reply:
the default constructor empties the vector, so why do you want to do this explicitly?